packages/bot-web-ui/src/stores/run-panel-store.js

```diff
@@ -10,18 +10,65 @@ export default class RunPanelStore {
         this.root_store = root_store;
         this.dbot = this.root_store.dbot;
         this.registerCoreReactions();
+
+        this.statistics_storage_key = 'statistics_cache';
+        this.empty_statistics = Object.freeze({
+            lost_contracts: 0,
+            number_of_runs: 0,
+            total_profit: 0,
+            total_payout: 0,
+            total_stake: 0,
+            won_contracts: 0,
+        });
+
+        this.disposeStatisticsListener = reaction(
+            () => this.statistics,
+            statistics => {
+                const { client } = this.root_store.core;
+                const stored_statistics = this.getSessionStorage(this.statistics_storage_key);
+
+                const new_statistics = { statistics };
+                stored_statistics[client.loginid] = new_statistics;
+
+                sessionStorage.setItem(this.statistics_storage_key, JSON.stringify(stored_statistics));
+            }
+        );
+
+        this.disposeSwitchAccountListener = reaction(
+            () => this.root_store.core.client.loginid,
+            () => {
+                const { core, transactions, journal, summary_card } = this.root_store;
+                const { client } = core;
+
+                if (client.is_logged_in) {
+                    this.statistics =
+                        this.getSessionStorage(this.statistics_storage_key)?.[client.loginid]?.statistics ??
+                        this.empty_statistics;
+
+                    journal.unfiltered_messages =
+                        this.getSessionStorage('journal_cache')?.[client.loginid]?.journal_messages ?? [];
+
+                    transactions.elements =
+                        this.getSessionStorage('transaction_cache')?.[client.loginid]?.transaction_elements ?? [];
```

Comment:
You're constantly getting the calls for a specific `loginid`, why not create a function that handles this for you which includes all of the checks that you're doing in these lines? It would make the code a lot cleaner.